packages/grafana-ui/src/components/Table/TableNG/Cells/TableCellNG.tsx

```diff
@@ -83,60 +86,66 @@ export function TableCellNG(props: TableCellNGProps) {
   }, [divWidthRef.current]); // eslint-disable-line react-hooks/exhaustive-deps
 
   // Common props for all cells
-  const commonProps = {
-    value,
-    field,
-    rowIdx,
-    justifyContent,
-  };
+  const commonProps = useMemo(
+    () => ({
+      value,
+      field,
+      rowIdx,
+      justifyContent,
+    }),
+    [value, field, rowIdx, justifyContent]
+  );
 
   // Get the correct cell type
-  let cell: ReactNode = null;
-  switch (cellType) {
-    case TableCellDisplayMode.Sparkline:
-      cell = <SparklineCell {...commonProps} theme={theme} timeRange={timeRange} width={divWidth} />;
-      break;
-    case TableCellDisplayMode.Gauge:
-    case TableCellDisplayMode.BasicGauge:
-    case TableCellDisplayMode.GradientGauge:
-    case TableCellDisplayMode.LcdGauge:
-      cell = <BarGaugeCell {...commonProps} theme={theme} timeRange={timeRange} height={height} width={divWidth} />;
-      break;
-    case TableCellDisplayMode.Image:
-      cell = <ImageCell {...commonProps} cellOptions={cellOptions} height={height} />;
-      break;
-    case TableCellDisplayMode.JSONView:
-      cell = <JSONCell {...commonProps} />;
-      break;
-    case TableCellDisplayMode.DataLinks:
-      cell = <DataLinksCell field={field} rowIdx={rowIdx} />;
-      break;
-    case TableCellDisplayMode.Actions:
-      cell = <ActionsCell actions={actions} />;
-      break;
-    case TableCellDisplayMode.Custom:
-      const CustomCellComponent: React.ComponentType<CustomCellRendererProps> = cellOptions.cellComponent;
-      cell = <CustomCellComponent field={field} value={value} rowIndex={rowIdx} frame={frame} />;
-      break;
-    case TableCellDisplayMode.Auto:
-    default:
-      // Handle auto cell type detection
-      if (field.type === FieldType.geo) {
-        cell = <GeoCell {...commonProps} height={height} />;
-      } else if (field.type === FieldType.frame) {
-        const firstValue = field.values[0];
-        if (isDataFrame(firstValue) && isTimeSeriesFrame(firstValue)) {
-          cell = <SparklineCell {...commonProps} theme={theme} timeRange={timeRange} width={divWidth} />;
-        } else {
+  const cell = useMemo(() => {
```

Comment:
```suggestion
  const cellContent = useMemo(() => {
```

Having the `const cell` and then inside the memo `let cell` is a bit confusing. Can we name the variable in line 100?

Reply:
good idea 👍